File: LeetCode_09/Solution_ZhongJing.kt
package LeetCode_09

/**
 * 日期：2023-07-02
 * <br>
 * 作者：仲景
 */
class Solution {
    fun isPalindrome(x: Int): Boolean {
        var num = x
        var res = 0

        while (num > 0) {
            res = res * 10 + (num % 10)
            num /= 10
        }

        return res == x
    }
}

File: LeetCode_09/LeetCode_09_02/Solution_ZhongJing.go
package main

// 作者：仲景
// 日期：2023-06-30
func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	num := x
	res := 0

	for num > 0 {
		res *= 10
		res += num % 10
		num /= 10
	}

	return res == x
}
